WcaOnRails/spec/features/sca_payments_spec.rb

```diff
@@ -1,64 +0,0 @@
-# frozen_string_literal: true
-
-require "rails_helper"
-
-RSpec.feature "Strong Customer Authentification payment" do
-  before :each do
-    # Enable CSRF protection just for these tests.
-    # See https://blog.tomoyukikashiro.me/post/test-csrf-in-feature-test-using-capybara/
-    allow_any_instance_of(ActionController::Base).to receive(:protect_against_forgery?).and_return(true)
-  end
-
-  context "on the 'register' page of a visible competition" do
-    let(:competition) { FactoryBot.create(:competition, :stripe_connected, :visible, :registration_open, events: Event.where(id: %w(222 333))) }
-    let!(:user) { FactoryBot.create(:user, :wca_id) }
-    let!(:registration) { FactoryBot.create(:registration, competition: competition, user: user) }
-    let!(:card) { FactoryBot.create(:credit_card, :sca_card) }
-
-    background do
-      sign_in user
-      visit competition_register_path(competition)
-    end
-
-    scenario "user pays with a 3D secure enabled card", js: true do
-      fill_confirm_and_expect(card, "test-source-authorize-3ds", "Your payment was successful")
-    end
-
-    it "user fails to complete the 3D secure challenge", js: true do
-      fill_confirm_and_expect(card, "test-source-fail-3ds", "We are unable to authenticate your payment method")
-    end
-  end
-end
-
-def fill_confirm_and_expect(card, button_id, message)
```

Comment:
Can we add tests for this back? Some possible ideas:

1. Bringing back a test where we type card details
2. Checking for some basic events like whether a `ready` event is emitted.
3. Making sure the Element is rendered with a payment method we expect

Reply:
Not for SCA specifically. Our testing framework is a _nightmare_ when it comes to iframes.
The reason why we had this test previously is that we were responsible for detecting and forwarding SCA events to Stripe manually. Now, Stripe does it for us so testing it in the way you described would be like testing Stripe's SDK is correct... Which I hope we shouldn't have to test at all.

Reply:
Missed that this was an SCA-specific file; I was thinking more about generic tests to make sure that the WCA site's integration is correct.

For instance, making sure we get the Payment Element rendering is an easy way to make sure that all the code that power the integration works. The goal wouldn't be to test Stripe's code or any particular edge cases that Stripe already handles. Instead, the goal would be an end-to-end test for the WCA site to create a payment intent, load the page, and render the Payment Element successfully.

Reply:
From the [Stripe documentation](https://stripe.com/docs/automated-testing):
> Frontend interfaces, like [Stripe Checkout](https://stripe.com/docs/payments/checkout) or the [Payment Element](https://stripe.com/docs/payments/payment-element), have security measures in place that prevent automated testing

This indeed makes it a pain in the butt to test, especially since the frontend is using iframes within iframes within iframes. I shall leave it at one simple setup test, that checks whether something renders.

Even creating a payment intent isn't necessarily a given anymore, seeing that we're using deferred PaymentIntents now. Checking those is really hard especially because of redirect URLs (our testing env isn't reachable from the outside world, and setting redirects to `if_required` goes against the "we don't want you to do automated frontend testing" policy of Stripe above)